lib/location/location.c

```diff
@@ -128,6 +128,7 @@ static void location_config_method_defaults_set(
 	if (method_type == LOCATION_METHOD_GNSS) {
 		method->gnss.timeout = 120;
 		method->gnss.accuracy = LOCATION_ACCURACY_NORMAL;
+		method->gnss.num_consecutive_fixes = 3;
```

Comment:
Is there at natural place for these defaults to be documented?

Reply:
I was also thinking about this earlier. The user can use the `location_config_method_defaults_set()` function to set the defaults, but the only place where the default values are listed is the code. It's probably not a good idea to document the defaults in the configuration structures, because the defaults get only applied if the function is called. The values could be described in the function description, but that would make it very long. So, I would say that there's no natural place to document these. Any good ideas? :)

Reply:
Documentation is the hardest part. But was thinking about num_consecutive_fixes config, do you think that the accuracy is better if having even more consecutive fixes although with the penalty of more pwr consumption? I.e. was thinking if that is worth of Kconfig item?

Reply:
3 is just the default value, the user can give a different value in the GNSS method configuration during runtime. I guess that should be fine?

Reply:
One benefit from having the defaults a Kconfig options is that they would then be documented :)

Reply:
Yes, I guess all the default values could be defined in Kconfig instead. But I think that could be in a way misleading, because the defaults would still only get applied if the `location_config_method_defaults_set()` function is called. It is not mandatory to use this function. If default values would be documented in Kconfig, one could assume that those are always used unless overridden in the configuration structure. We could of course document it in the Kconfig that those default values are only applied when the function is called, but I'm not sure if I like that very much either. :)

The configuration could have been implemented in a different way, I guess. For example, value zero in the configuration could have indicated that the default value (set and documented in Kconfig) should be used. This approach would have had a different problem, though: how to distinguish between value 0 and the intention to use the default. Or function `location_config_method_defaults_set()` could have had a different name and it could have been mandatory to call it to initialize the configuration structure.

I don't want to do bigger changes to the configuration in this PR, but we can do it in another PR if we can figure out a better way to do it.

Reply:
Yeah, this isn't straight/forward. No problem leaving as is and revisit later

Reply:
As discussed with @jtguggedal offline, I will create a new PR to document the default values. The defaults can be set in Kconfig instead, it just has to be mentioned that location_config_method_defaults_set() still needs to be called to apply the default values.